nullaway/src/main/java/com/uber/nullaway/dataflow/cfg/NullAwayCFGBuilder.java

```diff
@@ -0,0 +1,185 @@
+package com.uber.nullaway.dataflow.cfg;
+
+import com.google.common.base.Preconditions;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.ThrowTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.TreeVisitor;
+import com.sun.source.util.TreePath;
+import com.uber.nullaway.handlers.Handler;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.nullaway.dataflow.cfg.ControlFlowGraph;
+import org.checkerframework.nullaway.dataflow.cfg.UnderlyingAST;
+import org.checkerframework.nullaway.dataflow.cfg.builder.CFGBuilder;
+import org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseOne;
+import org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseThree;
+import org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseTwo;
+import org.checkerframework.nullaway.dataflow.cfg.builder.ConditionalJump;
+import org.checkerframework.nullaway.dataflow.cfg.builder.ExtendedNode;
+import org.checkerframework.nullaway.dataflow.cfg.builder.Label;
+import org.checkerframework.nullaway.dataflow.cfg.builder.PhaseOneResult;
+import org.checkerframework.nullaway.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.nullaway.dataflow.cfg.node.Node;
+import org.checkerframework.nullaway.dataflow.cfg.node.ThrowNode;
+import org.checkerframework.nullaway.javacutil.AnnotationProvider;
+import org.checkerframework.nullaway.javacutil.BasicAnnotationProvider;
+import org.checkerframework.nullaway.javacutil.trees.TreeBuilder;
+
+/**
+ * A NullAway specific CFGBuilder subclass, which allows to more directly control the AST to CFG
+ * translation performed by the checker framework.
+ *
+ * <p>This holds the static method {@link #build(TreePath, UnderlyingAST, boolean, boolean,
+ * ProcessingEnvironment, Handler)}, called to perform the CFG translation, and the class {@link
+ * NullAwayCFGTranslationPhaseOne}, which extends {@link CFGTranslationPhaseOne} and adds hooks for
+ * the NullAway handlers mechanism and some utility methods.
+ */
+public final class NullAwayCFGBuilder extends CFGBuilder {
+
+  /** This class should never be instantiated. */
+  private NullAwayCFGBuilder() {}
+
+  /**
+   * This static method produces a new CFG representation given a method's (or lambda/initializer)
+   * body.
+   *
+   * <p>It is analogous to {@link CFGBuilder#build(TreePath, UnderlyingAST, boolean, boolean,
+   * ProcessingEnvironment)}, but it also takes a handler to be called at specific extention points
+   * during the CFG translation.
+   *
+   * @param bodyPath the TreePath to the body of the method, lambda, or initializer.
+   * @param underlyingAST the AST that underlies the control frow graph
+   * @param assumeAssertionsEnabled can assertions be assumed to be disabled?
+   * @param assumeAssertionsDisabled can assertions be assumed to be enabled?
```

Comment:
We could hard-code some of these to what NullAway already uses in every case, but maybe in the future we can see a reason to change them, and it doesn't feel like much extra work to expose them to caller of `NullAwayCFGBuilder`

Reply:
Agreed